src/index.ts

```diff
@@ -1,173 +1,231 @@
+  }
 
-  return {
-    destroy: function () {
-      // remove event listeners
-      vgView.removeEventListener('mouseover.tooltipInit');
-      vgView.removeEventListener('mousemove.tooltipUpdate');
-      vgView.removeEventListener('mouseout.tooltipRemove');
+  if (isString(value)) {
+    return value;
+  }
 
-      cancelPromise(); // clear tooltip promise
-    }
-  };
-}
+  if (isObject(value)) {
+    let content = '';
 
-function start(vgView: VgView, options: Option) {
-  // initialize tooltip with item data and options on mouse over
-  vgView.addEventListener('mouseover.tooltipInit', function (event: MouseEvent, item: Scenegraph) {
-    if (shouldShowTooltip(item)) {
-      // clear existing promise because mouse can only point at one thing at a time
-      cancelPromise();
-
-      // make a new promise with time delay for tooltip
-      tooltipPromise = window.setTimeout(function () {
-        init(event, item, options);
-      }, options.delay || DELAY);
-    }
-  });
+    const { title, ...rest } = value as any;
 
-  // update tooltip position on mouse move
-  // (important for large marks e.g. bars)
-  vgView.addEventListener('mousemove.tooltipUpdate', function (event: MouseEvent, item: Scenegraph) {
-    if (shouldShowTooltip(item) && tooltipActive) {
-      update(event, item, options);
+    if (title) {
+      content += `<h2>${title}</h2>`;
     }
-  });
-
-  // clear tooltip on mouse out
-  vgView.addEventListener('mouseout.tooltipRemove', function (event: MouseEvent, item: Scenegraph) {
-    if (shouldShowTooltip(item)) {
-      cancelPromise();
 
-      if (tooltipActive) {
-        clear(event, item, options);
+    content += '<table>';
+    for (const key of Object.keys(rest)) {
+      let val = (rest as any)[key];
+      if (isObject(val)) {
+        val = stringify(val);
       }
+
+      content += `<tr><td class="key">${key}:</td><td class="value">${val}</td></tr>`;
     }
-  });
-}
+    content += `</table>`;
+
+    return content;
+  }
 
-/* Cancel tooltip promise */
-function cancelPromise() {
-  /* We don't check if tooltipPromise is valid because passing
-   an invalid ID to clearTimeout does not have any effect
-   (and doesn't throw an exception). */
-  window.clearTimeout(tooltipPromise);
-  tooltipPromise = undefined;
+  return `${value}`;
 }
 
-/* Initialize tooltip with data */
-function init(event: MouseEvent, item: Scenegraph, options: Option): void {
-  // get tooltip HTML placeholder
-  const tooltipPlaceholder = getTooltipPlaceholder();
+/**
+ * Attach a tooltip handler to the view.
+ */
+export default function(view: View, opt?: Options) {
+  let visible = false;
+
+  const options = { ...DEFAULT_OPTIONS, ...opt };
 
-  // prepare data for tooltip
-  const tooltipData = getTooltipData(item, options);
-  if (!tooltipData || tooltipData.length === 0) {
-    return undefined;
+  init(options);
+
+  function hideTooltip() {
+    tooltipElement.classList.remove('visible', `${options.theme}-theme`);
```

Comment:
tbh, I think tooltipElement should be a per-instance variable, returned from the init() function.